src/dft/backends/mklcpu/forward.cpp

```diff
@@ -153,8 +153,8 @@ ONEMKL_EXPORT void compute_forward(descriptor_type &desc, sycl::buffer<input_typ
 
     cpu_queue.submit([&](sycl::handler &cgh) {
         auto desc_acc = mklcpu_desc_buffer.template get_access<sycl::access::mode::read>(cgh);
-        auto in_acc = in.template get_access<sycl::access::mode::read>(cgh);
-        auto out_acc = out.template get_access<sycl::access::mode::write>(cgh);
+        auto in_acc = in.template get_access<sycl::access::mode::read_write>(cgh);
+        auto out_acc = out.template get_access<sycl::access::mode::read_write>(cgh);
 
         detail::host_task<class host_kernel_outofplace>(cgh, [=]() {
             DFT_ERROR status = DftiComputeForward(desc_acc[detail::DIR::fwd], in_acc.get_pointer(),
```

Comment:
I would not agreed with the use of `const_cast`. If `DftiComputeXXX` want to guarantee that their API doesn't modify the values of the input, then the library authors can change the function signature. Otherwise they could change the implementation to modify the input in the future and we might not notice!

Reply:
@FMarno the old variadic C API cannot give any programatic gurantee because the first (and second for split) argument can be treated as inout and sometimes as unchangable inputs depending on the runtime argument list. So no we can't *easily* change the function signature! we gurantee the OOP computation preserves input but as a documented promise.

Reply:
It's not ideal but if we can't modify the C API then it's best to use a `const_cast` IMO. Is it something that could be revisited eventually? It sounds useful for users if that API is meant to be used in C++ applications.

Reply:
Yeah, the root of the problem is that this API is targeting C applications. We used the C APIs here instead of the C++ APIs to avoid the complications of our C++ API being essentially the same as the interface API here. Maybe @anantsrivastava30 can say more about whether the GPU enabling cleared up enough of those complications to consider revisiting the CPU enabling.

Reply:
From what I understand, we enabled the CPU backend using the classical API because we wanted to demonstrate that different APIs can be used to create the device specific interfaces, so we will not be, using the C++ API for CPU. 
I believe this use case of using c_api from a c++ source is rare and should not warrant a rather big (closed-source) oneMKL change. 
We can revisit this later if the fact that the c_api is rarely used from a C++ source changes, for now to get the CI working let's resolve this by using const_cast.

Reply:
Alright, but I think we should at least have a link to the place in the documentation where it says that it is guaranteed to not modify the value of input for out-of-place transforms.

Reply:
Thanks for the extra context about the CPU enabling @anantsrivastava30. Here's the part of the C documentation showing you'd have to explicitly set [DFTI_DESTROY_INPUT](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-c/2023-1/dfti-destroy-input.html) to `DFTI_ALLOW` for the out-of-place input to be changed.